src/app/components/log-in/log-in.component.ts

```diff
@@ -1,22 +1,67 @@
-import { Component } from '@angular/core';
-import { FormControl, FormGroup, ReactiveFormsModule } from '@angular/forms';
+import { Component, inject } from '@angular/core';
+import {
+  FormControl,
+  FormGroup,
+  ReactiveFormsModule,
+  Validators,
+} from '@angular/forms';
+import { Router, RouterLink } from '@angular/router';
+import { NgIconComponent, provideIcons } from '@ng-icons/core';
+import {
+  heroEye,
+  heroEyeSlash,
+  heroLockClosed,
+  heroUser,
+} from '@ng-icons/heroicons/outline';
+import { AuthService } from '../../services/auth.service';
 import { BackgroundComponent } from '../shared-components/background/background.component';
 
 @Component({
   selector: 'app-log-in',
   standalone: true,
-  imports: [ReactiveFormsModule, BackgroundComponent],
+  imports: [
+    ReactiveFormsModule,
+    BackgroundComponent,
+    NgIconComponent,
+    RouterLink,
+  ],
   templateUrl: './log-in.component.html',
   styleUrl: './log-in.component.scss',
+  viewProviders: [
+    provideIcons({ heroUser, heroLockClosed, heroEye, heroEyeSlash }),
+  ],
 })
 export class LogInComponent {
+  authService = inject(AuthService);
+  router = inject(Router);
+  showPassword: boolean = true;
+
   loginForm = new FormGroup({
-    username: new FormControl(''),
-    password: new FormControl(''),
+    email: new FormControl('', [Validators.required]),
+    password: new FormControl('', [Validators.required]),
   });
 
+  errorMessage: string | null = null;
+
   onSubmit() {
-    // TODO: Use EventEmitter with form value
-    console.warn(this.loginForm.value);
+    // TODO: when this is implememted you need to have a spinner to indicate that the form is being submitted
+    console.log(this.loginForm.value);
+    const rawForm = this.loginForm.getRawValue();
+
+    this.authService
+      .login(rawForm.email ?? '', rawForm.password ?? '')
+      .subscribe({
```

Comment:
Remember to destroy subscriptions.

Reply:
yeah, spoke with Tim, this will change when changing to NgRx, which I am currently implementing